vendor/github.com/danwinship/knftables/nftables.go

```diff
@@ -0,0 +1,444 @@
+			table:  table,
+		},
+
+		exec: execer,
+	}
+
+	nft.path, err = nft.exec.LookPath("nft")
+	if err != nil {
+		return nil, fmt.Errorf("could not find nftables binary: %w", err)
+	}
+
+	cmd := exec.Command(nft.path, "--check", "add", "table", string(nft.family), nft.table)
+	_, err = nft.exec.Run(cmd)
+	if err != nil {
+		return nil, fmt.Errorf("could not run nftables command: %w", err)
+	}
+
+	cmd = exec.Command(nft.path, "--check", "add", "table", string(nft.family), nft.table,
+		"{", "comment", `"test"`, "}",
+	)
+	_, err = nft.exec.Run(cmd)
+	if err != nil {
+		nft.noObjectComments = true
+	}
+
+	return nft, nil
+}
+
+// New creates a new nftables.Interface for interacting with the given table. If nftables
+// is not available/usable on the current host, it will return an error.
+func New(family Family, table string) (Interface, error) {
+	return newInternal(family, table, realExec{})
+}
+
+// NewTransaction is part of Interface
+func (nft *realNFTables) NewTransaction() *Transaction {
+	return &Transaction{nftContext: &nft.nftContext}
+}
+
+// Run is part of Interface
+func (nft *realNFTables) Run(ctx context.Context, tx *Transaction) error {
+	if tx.err != nil {
+		return tx.err
+	}
+
+	buf, err := tx.asCommandBuf()
+	if err != nil {
+		return err
+	}
+
+	cmd := exec.CommandContext(ctx, nft.path, "-f", "-")
+	cmd.Stdin = buf
+	_, err = nft.exec.Run(cmd)
+	return err
+}
+
+// jsonVal looks up key in json; if it exists and is of type T, it returns (json[key], true).
+// Otherwise it returns (_, false).
+func jsonVal[T any](json map[string]interface{}, key string) (T, bool) {
+	if ifVal, exists := json[key]; exists {
+		tVal, ok := ifVal.(T)
+		return tVal, ok
+	} else {
+		var zero T
+		return zero, false
+	}
+}
+
+// getJSONObjects takes the output of "nft -j list", validates it, and returns an array
+// of just the objects of objectType.
+func getJSONObjects(listOutput, objectType string) ([]map[string]interface{}, error) {
+	// listOutput should contain JSON looking like:
+	//
+	// {
+	//   "nftables": [
+	//     {
+	//       "metainfo": {
+	//         "json_schema_version": 1,
+	//         ...
+	//       }
+	//     },
+	//     {
+	//       "chain": {
+	//         "family": "ip",
+	//         "table": "kube-proxy",
+	//         "name": "KUBE-SERVICES",
+	//         "handle": 3
+	//       }
+	//     },
+	//     {
+	//       "chain": {
+	//         "family": "ip",
+	//         "table": "kube-proxy",
+	//         "name": "KUBE-NODEPORTS",
+	//         "handle": 4
+	//       }
+	//     },
+	//     ...
+	//   ]
+	// }
+	//
+	// In this case, given objectType "chain", we would return
+	//
+	// [
+	//   {
+	//     "family": "ip",
+	//     "table": "kube-proxy",
+	//     "name": "KUBE-SERVICES",
+	//     "handle": 3
+	//   },
+	//   {
+	//     "family": "ip",
+	//     "table": "kube-proxy",
+	//     "name": "KUBE-NODEPORTS",
+	//     "handle": 4
+	//   },
+	//   ...
+	// ]
+
+	jsonResult := map[string][]map[string]map[string]interface{}{}
```

Comment:
is this consistent and reliable?

Reply:
what, haven't you ever seen a `map[string][]map[string]map[string]interface{}{}` before??

If you mean, is the JSON syntax consistent and reliable, [yes](https://manpages.debian.org/testing/libnftables1/libnftables-json.5.en.html). Note that we check the `json_schema_version` value too in case that changes.

If you mean, is that code written in the best way it could be, probably not. The JSON schema was unfortunately designed by someone using it from python, so it works well in a dynamically typed language, but terribly in a statically typed language. I would like to find some better way to deal with it...

Reply:
> what, haven't you ever seen a `map[string][]map[string]map[string]interface{}{}` before??

I didn't and  I only found one reference here https://grep.app/search?q=map%5Bstring%5D%5B%5Dmap%5Bstring%5Dmap%5Bstring%5Dinterface%7B%7D%7B%7D  and 7 in github https://github.com/search?q=map%5Bstring%5D%5B%5Dmap%5Bstring%5Dmap%5Bstring%5Dinterface%7B%7D%7B%7D&type=code

not my area of knowledge, but is not this something that Unstructured is supposed to solve https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured#Unstructured?

Reply:
> I didn't

That was a rhetorical question. It's obviously terrible. However, it _is_ an accurate representation of the top few levels of the nftables JSON representation.

> is not this something that Unstructured is supposed to solve

No, "Unstructured objects still have functioning TypeMeta features-- kind, version, etc."

Reply:
> That was a rhetorical question
lol, got it know

Reply:
Barf!